portal-impl/src/com/liferay/portal/zip/ZipWriterImpl.java

```diff
@@ -73,7 +73,7 @@ public ZipWriterImpl(File file) {
 
 		URI fileURI = _file.toURI();
 
-		_uri = URI.create("jar:file:" + fileURI.getPath());
+		_uri = URI.create("jar:file:" + fileURI.getRawPath());
```

Comment:
@JanisZhang Please don't mention `URISyntaxException` in the commit message, but focus on what the correct way is.

I pulled this PR and modified it to:

> LPS-150847 Should use URI.getRawPath() instead of URI.getPath() when creating a new URI from an existing URI instance. When creating a URI, illegal characters in the path must be encoded (File.toURI() does this). When getting the path, URI.getRawPath() returns the encoded path as is, but URI.getPath() returns the decoded path, in which the illegal characters are restored.

Please change it on your side and force push. Thanks.